Network/Wai/Middleware/Static.hs

```diff
@@ -234,10 +234,15 @@ unsafeStaticPolicy' cc = unsafeStaticPolicyWithOptions (defaultOptions { cacheCo
 -- this has no policies enabled by default and is hence insecure. Takes 'Options'.
 unsafeStaticPolicyWithOptions :: Options -> Policy -> Middleware
 unsafeStaticPolicyWithOptions options p app req callback =
-    maybe (app req callback)
-          (\fp ->
-               do exists <- liftIO $ doesFileExist fp
-                  if exists
+    maybe serveUpstream tryStaticFile mCandidateFile
+    where
+      serveUpstream :: IO ResponseReceived
+      serveUpstream = app req callback
+
+      tryStaticFile :: FilePath -> IO ResponseReceived
+      tryStaticFile fp = do
+          exists <- liftIO $ doesFileExist fp
```

Comment:
I don't think this `liftIO` is needed here. (This was present in the original code, but I only noticed this now.)

Reply:
Hey, this is nice! It allows us to get rid of the `mtl` dependency as well (which should have been `transformers` in the first place).

Reply:
Done.